File: VCP.lua
message = "THEJAVAPROGRAMMER"
keyText = "NEERAJ"

--msg = message.split("")
--key = keyText.split("")
--msgLen = msg.length

msg = {}
for i = 1,#message do
	msg[i] =  message:sub(i,i)
end

key = {}
for i=1,#keyText do
	key[i] = keyText:sub(i,i)
end

msgLen = #message

newKey = {}
encryptedMsg = {}
decryptedMsg = {}


j = 1
for i = 1, msgLen do
	if j == #keyText+1 then
		j = 1
	end
	newKey[i] = key[j];
	j = j + 1
end


for i = 1,msgLen do
	encryptedMsg[i] = string.char(((string.byte(msg[i]) + string.byte(newKey[i])) % 26) + string.byte("A"))
	--encryptedMsg[i] = string.char(((msg[i].charCodeAt(0) + newKey[i].charCodeAt(0)) % 26) + 'A'.charCodeAt(0))
end
for i = 1,msgLen do
	decryptedMsg[i] = string.char((((string.byte(encryptedMsg[i]) - string.byte(newKey[i])) + 26) % 26) + string.byte("A"))
	--decryptedMsg[i] = string.char(((((encryptedMsg[i].charCodeAt(0) - newKey[i].charCodeAt(0)) + 26) % 26) + 'A'.charCodeAt(0)))
end

function tablePrint(l)
	str = ""
	for i,v in pairs(l) do
		--io.write(v)
		str = str..v
	end
	return str
end


print("Original Message: "..tablePrint(msg))
print("Key: "..tablePrint(key))
print("Generated Key: "..tablePrint(newKey))
print("Encrypted Message: "..tablePrint(encryptedMsg))
print("Decrytped Message: "..tablePrint(decryptedMsg))

File: VCP.swift
import Foundation
func Int_to_Char(i: UInt32) -> Character{
	let u = UnicodeScalar(i)
	let c = Character(u!)
	return c
}

let message = "THESWIFTPROGRAMMER"
let keyText = "NEERAJ"

let msg = Array(message)
let key = Array(keyText)
let msgLen = msg.count

var newKey = [String]()
var encryptedMsg = [String]()
var decryptedMsg = [String]()

var j = 0
for _ in 0...msgLen-1{
	if j == key.count{
		j = 0
	}
	
	newKey.append(String(key[j]))
	j = j+1
}

for i in 0...msgLen-1{
	encryptedMsg.append(String(Int_to_Char(i: (((UnicodeScalar(String(msg[i]))!.value + UnicodeScalar(String(newKey[i]))!.value) % 26) + UnicodeScalar("A")!.value))))
}

for i in 0...msgLen-1{
	decryptedMsg.append(String(Int_to_Char(i: UInt32((((Int(UnicodeScalar(encryptedMsg[i])!.value) - Int(UnicodeScalar(newKey[i])!.value)) + Int(26)) % 26) + Int(UnicodeScalar("A")!.value)))))
	
}
// Result has 3 strings.
print("Original Message: \(message)")
print("Key: \(keyText)")
print("Generated Key: " + newKey.joined(separator: ""))
print("Encrypted Message: " + encryptedMsg.joined(separator: ""))
print("Decrypted Message: " + decryptedMsg.joined(separator: ""))